AppCenter/AppCenter/Internals/Util/MSDispatcherUtil.h

```diff
@@ -3,8 +3,28 @@
 
 #import <Foundation/Foundation.h>
 
+#define ARRAY_FROM_ARGS(...)                                                                                                               \
+  ({                                                                                                                                       \
+    NSMutableArray *initedArray = [NSMutableArray arrayWithObjects:[NSNull null], ##__VA_ARGS__, nil];                                     \
```

Comment:
A problem here is if you pass a nil value in the middle of the arguments, it'll ignore any other arguments. For instance `NSArray *array = ARRAY_FROM_ARGS(@1, nil, @2);` here `@2` will be ignored. Although not sure how to fix it 🤔.

Reply:
I think we can't do this with the usual C and Objective-C tools. We can't cycle through parameters using a variadic func and change them. We can try to impose restrictions on the function in this case. Pass an NSNull value instead of nil. In this case we can find out what positions NSNull is in and convert them to nil.

Reply:
Wait, isn't this solving our issues here:
```
#define MS_DISPATCH_SELECTOR(declaration, object, selectorName, ...)({ \
(declaration [object methodForSelector:NSSelectorFromString(selectorName)])(object, NSSelectorFromString(selectorName), ##__VA_ARGS__); \
})
```
What was problematic was guessing the types of the parameters and returned value but we can actually ask for them. This is how it would work for instance:
`    MS_DISPATCH_SELECTOR((void (*)(id, SEL, typeof(windowScene))), window, @"setWindowScene:", windowScene);`
Just a one liner.